Input is sometimes needed for  visualization

riemarc opened the pull request:
See 08e1c4a: Assuming that there is no other/better way, we need a feature like this.

riemarc committed:
HACK we need the input for visualization

cklb committed:
Added try/except clause on module init

cklb committed:
Merge branch 'master' into develop

cklb commented:
There is indeed no other way to get those information. Can you think of additional simulation results you might need for other simulations?
The solution I'm currently thinking of would be a to move this functionality into the visualizer base class and thus providing it with the complete set of simulation results. Next, the visualizer is called with the current time and selects values as needed. For compatibility reasons and easier usage, interpolators for input and state will be provided by the visualizer base class.
I'll build an example so that we can discuss further details.

cklb commented:
My idea can be found in  the branch `visualizer_extension`.
Since the present examples do not need the model input for visualization, for now I'm not able to add a real example but the doc should explain enough.

cklb committed:
Introduced a modular approach for visualizers

This way, the visualizer can access all simulation results if needed
while keeping compatibility to the existing implementations.

riemarc committed:
Add code for ball on plate from fabians ba thesis

riemarc committed:
Activate input interpolator feature

riemarc commented:
Looks good! I think that's a good design decision. Only the interface needs to be generalized. I added an example from @FabianBOP, but I dont know how to port this to the current develepment version of pymoskito.

cklb committed:
Merge remote-tracking branch 'umit/stand_fabian_pc' into extended_visu

cklb committed:
Adapted visualizer interface

cklb committed:
Started adaption of new example

cklb commented:
Merged into the current develop and applied changes to the example.
Please remember to remove all remaining lines that have nothing to do with the new example and add some information about the system.

cklb committed:
Started cleanup of ball on plate example
